addons/escoria-core/game/inputs_manager.gd

```diff
@@ -102,20 +114,30 @@ func register_background(background: ESCBackground):
 	)
 
 
-# Registers a function that has the first chance to process
-# _unhandled_input in an ESCBackground when in GAME_STATE.DEFAULT.
-# The callback receives the InputEvent as its only argument and
-# must represent a bool indicating whether it handled the event.
-# If true, ESCBackground will not attempt to process the event.
+# Registers a function that can be used to intercept and process input events.
+# `callback` must have the following signature:
 #
-# Note that funcref() is helpful in defining a callback function.
-func register_custom_background_input_handler(callback) -> void:
-	custom_background_input_handler = callback
+# (event: InputEvent, is_default_state: bool) -> bool
+#
+# where
+#
+# - event: The event to process
+# - is_default_state: Whether the current state is escoria.GAME_STATE.DEFAULT
+# - returns whether the function processed the event. If so,
+#   `get_tree().set_input_as_handled()` will be called.
+#
+# #### Parameters
+# - callback: Function satisfying the above contract
+func register_custom_input_handler(callback) -> void:
+	custom_input_handler = callback
 
 
-func handle_background_input(event: InputEvent, is_default_state: bool) -> bool:
-	if custom_background_input_handler:
-		return custom_background_input_handler.call_func(event, is_default_state)
+# If a callback was specified via `register_custom_input_handler()`,
```

Comment:
@BHSDuncan I'm not sure exactly what you mean, but I believe the challenge is we are documenting a parameter (whose type is function) as well as the contract of the function, and I didn't find an existing instance of how that is done handy, so I had to make something up. Since only `callback` is the parameter, is the only thing explicitly listed under `#### Parameters`

Reply:
(I was looking in the other view and I think the line numbers shifted since the comment was made...)

Reply:
OK, I believe this is fixed now!

Perhaps this raises the question of whether the call to `get_tree().set_input_as_handled()` should be moved from `esc_background.gd` to `inputs_manager.gd`?

Reply:
Sorry, I probably wasn't clear enough (wasn't sure how to select multiple lines).

I tried to highlight the first docstring for `try_customer_input_handler` to indicate that entire comment block: I was hoping you could format it more like the other functions, i.e.

```
<description>

<params>

<returns>
```

But good improvement anyway!

Reply:
Ugh, I hate Git so much...I forgot to add to the index before that last update...

Reply:
OK, the requested bits are there now.